src/modules/asset/sagas.ts

```diff
@@ -6,57 +6,73 @@ import {
+import { sleep } from 'decentraland-commons/dist/utils'
 
-export function* assetSaga() {
+export function* assetSaga(client: BuilderClient) {
   yield takeLatest(LOAD_COLLECTIBLES_REQUEST, handleLoadCollectibles)
   yield takeLatest(CONNECT_WALLET_SUCCESS, handleConnectWallet)
-}
 
-function* handleConnectWallet() {
-  yield put(loadCollectiblesRequest())
-}
+  function* handleConnectWallet() {
+    yield put(loadCollectiblesRequest())
+  }
 
-function* handleLoadCollectibles(_: LoadCollectiblesRequestAction) {
-  const address: string | null = yield select(getAddress)
+  function* handleLoadCollectibles(_: LoadCollectiblesRequestAction) {
+    const address: string | null = yield select(getAddress)
 
-  try {
-    if (!address) {
-      throw new Error(`Invalid address: ${address}`)
+    try {
+      if (!address) {
+        throw new Error(`Invalid address: ${address}`)
+      }
+      const assets: Asset[] = []
+      const serverNFTs: NFT[] = yield call(getNFTs, address)
+      for (const openseaAsset of serverNFTs) {
+        const uri = `ethereum://${openseaAsset.contract.address}/${openseaAsset.tokenId}`
+        assets.push({
+          assetPackId: COLLECTIBLE_ASSET_PACK_ID,
+          id: uri,
+          tags: [],
+          category: openseaAsset.contract.name,
+          contents: {},
+          name: openseaAsset.name || '',
+          model: uri,
+          script: null,
+          thumbnail: openseaAsset.imageThumbnailUrl || TRANSPARENT_PIXEL,
+          metrics: {
+            triangles: 0,
+            materials: 0,
+            meshes: 0,
+            bodies: 0,
+            entities: 0,
+            textures: 0
+          },
+          parameters: [],
+          actions: []
+        })
+      }
+      yield put(loadCollectiblesSuccess(assets))
+    } catch (error) {
+      yield put(loadCollectiblesFailure(error.message))
     }
-    const assets: Asset[] = []
-    const openseaAssets: OpenSeaAsset[] = yield call(() => opensea.fetchAssets(address))
-    for (const openseaAsset of openseaAssets) {
-      const uri = `ethereum://${openseaAsset.asset_contract.address}/${openseaAsset.token_id}`
-      assets.push({
-        assetPackId: COLLECTIBLE_ASSET_PACK_ID,
-        id: uri,
-        tags: [],
-        category: openseaAsset.asset_contract.name,
-        contents: {},
-        name: openseaAsset.name || '',
-        model: uri,
-        script: null,
-        thumbnail: openseaAsset.image_thumbnail_url || TRANSPARENT_PIXEL,
-        metrics: {
-          triangles: 0,
-          materials: 0,
-          meshes: 0,
-          bodies: 0,
-          entities: 0,
-          textures: 0
-        },
-        parameters: [],
-        actions: []
-      })
+  }
+
+  async function getNFTs(owner: string, cursor?: string): Promise<NFT[]> {
+    const response = await client.getNFTs({ owner, cursor })
+
+    const { next, nfts } = response
+
+    if (next) {
+      await sleep(400)
```

Comment:
Question: why's this necessary? I see that the old implementation had a `sleep(400)` but we keeping it here for a particular reason?
seems like we could just hit the API right?

Reply:
I don't know what the reason behind the sleep was. Just kept it there as the backend is still calling the same opensea endpoint. Yet I think its safe to remove it.